src/test/java/org/zalando/nakadi/service/subscription/state/PartitionDataTest.java

```diff
@@ -1,89 +1,74 @@
+import org.zalando.nakadi.repository.kafka.KafkaCursor;
 import static java.lang.System.currentTimeMillis;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
 
 public class PartitionDataTest {
 
+    private static Timeline fakeTimeline = mock(Timeline.class);
+
+    private static NakadiCursor createCursor(final long offset) {
+        return new KafkaCursor("x", 0, offset).toNakadiCursor(fakeTimeline);
+    }
+
     @Test
     public void onNewOffsetsShouldSupportRollback() {
-        final PartitionData pd = new PartitionData(null, 100L);
-        final PartitionData.CommitResult cr = pd.onCommitOffset(90L);
+        final PartitionData pd = new PartitionData(null, createCursor(100L));
+        final PartitionData.CommitResult cr = pd.onCommitOffset(createCursor(90L));
 
         assertEquals(0L, cr.committedCount);
         assertEquals(true, cr.seekOnKafka);
-        assertEquals(90L, pd.getSentOffset());
+        assertEquals(90L, Long.parseLong(pd.getSentOffset().getOffset()));
         assertEquals(0L, pd.getUnconfirmed());
     }
 
     @Test
     public void onNewOffsetsShouldSupportCommitInFuture() {
-        final PartitionData pd = new PartitionData(null, 100L);
-        final PartitionData.CommitResult cr = pd.onCommitOffset(110L);
+        final PartitionData pd = new PartitionData(null, createCursor(100L));
+        final PartitionData.CommitResult cr = pd.onCommitOffset(createCursor(110L));
 
-        assertEquals(10L, cr.committedCount);
+        assertEquals(0L, cr.committedCount);
         assertEquals(true, cr.seekOnKafka);
-        assertEquals(110L, pd.getSentOffset());
+        assertEquals(110L, Long.parseLong(pd.getSentOffset().getOffset()));
         assertEquals(0L, pd.getUnconfirmed());
     }
 
     @Test
     public void normalOperationShouldNotReconfigureKafkaConsumer() {
-        final PartitionData pd = new PartitionData(null, 100L);
+        final PartitionData pd = new PartitionData(null, createCursor(100L));
         for (long i = 0; i < 100; ++i) {
-            pd.addEventFromKafka(100L + i + 1, "test_" + i);
+            pd.addEvent(new ConsumedEvent("test_" + i, createCursor(100L + i + 1)));
         }
         // Now say to it that it was sent
         pd.takeEventsToStream(currentTimeMillis(), 1000, 0L);
         assertEquals(100L, pd.getUnconfirmed());
         for (long i = 0; i < 10; ++i) {
-            final PartitionData.CommitResult cr = pd.onCommitOffset(110L + i * 10L);
+            final PartitionData.CommitResult cr = pd.onCommitOffset(createCursor(110L + i * 10L));
             assertEquals(10L, cr.committedCount);
             assertFalse(cr.seekOnKafka);
             assertEquals(90L - i * 10L, pd.getUnconfirmed());
         }
     }
 
     @Test
-    public void eventsMustBeReturnedInGuaranteedOrder() {
```

Comment:
@rcillo 
The content of this test present all over subscription api tests (checking order).
From the other hand - it is very hard to reproduce this test in new code, case it requires much more actions to be done. 
That is why I decided that this test can be removed.